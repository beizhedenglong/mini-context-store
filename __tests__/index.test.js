/* eslint-disable react/prop-types */
import React from 'react'
import renderer from 'react-test-renderer'
import createStore from '../src/index'


describe('createStore', () => {
  const state = { number: 0 }
  const actions = { onAdd: num => ({ number }) => ({ number: number + num }) }
  const { Provider, connect } = createStore(
    state,
    actions,
  )
  let Add = ({ onAdd, number }) => (
    <div>
      {number}
      <button onClick={() => onAdd(1)}>+1</button>
    </div>
  )
  const identityMock = jest.fn(x => x)
  Add = connect(identityMock, identityMock)(Add)


  const App = () => (<Provider>hello, world!<Add /></Provider>)
  const tree = renderer.create(<App />)
  const provider = tree.root.findByType(Provider).instance

  test('renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('should add state to provider', () => {
    expect(provider.state).toBe(state)
    expect(provider.state.number).toBe(0)
  })

  test('actions', () => {
    const button = tree.root.findByType('button')
    expect(identityMock.mock.calls.length).toBe(2)
    button.props.onClick()
    expect(tree.toJSON()).toMatchSnapshot()
    expect(provider.state.number).toBe(1)
    expect(identityMock.mock.calls.length).toBe(4)
    expect(identityMock.mock.calls[identityMock.mock.calls.length - 2]).toEqual([provider.state])
  })
})
