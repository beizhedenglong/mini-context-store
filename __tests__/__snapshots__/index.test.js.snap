// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createStore handlers 1`] = `
Array [
  "hello, world!",
  <div>
    1
    <button
      onClick={[Function]}
    >
      +1
    </button>
  </div>,
]
`;

exports[`createStore renders correctly 1`] = `
Array [
  "hello, world!",
  <div>
    0
    <button
      onClick={[Function]}
    >
      +1
    </button>
  </div>,
]
`;
